include/envoy/api/api.h

```diff
@@ -57,9 +57,9 @@ class Api {
   virtual TimeSource& timeSource() PURE;
 
   /**
-   * @return a constant reference to the root Stats::Scope
+   * @return a reference to the root Stats::Scope
    */
-  virtual const Stats::Scope& rootScope() PURE;
```

Comment:
I had to drop the const so I could pass from the dispatcher into the resolver's constructor and call createScope. Any reservations with this change?

Reply:
I think the theoretical downside of this change is it would allow new scopes to be created much more widely than they are today. I'm guessing that is why this was the way it was before. I'm not going to throw up a theoretical blocker to this, but it would be nice to do this another way if possible. If not, this is fine.

Reply:
Yeah, that makes sense. Unfortunately, I don't see an easy way to get the resolver a scope from within the dispatcher other than the Api interface.